streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java

```diff
@@ -155,4 +163,20 @@ public void createTopic(final String topic,
     public void deleteTopic(final String topic) {
         brokers[0].deleteTopic(topic);
     }
+
+    public void waitForTopics(final StreamsConfig config, final String...topics) throws InterruptedException, IOException {
+        final StreamsKafkaClient streamsKafkaClient = new StreamsKafkaClient(config);
```

Comment:
Instead of using a separate client, could we migrate the scala `TestUtils.waitUntilMetadataIsPropagated` code, that looks into `broker.apis.metadataCache` directly?